Segfault on scaleway_iam_application import

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform Version

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->
```
❯ terraform -v
Terraform v1.8.2
on linux_amd64
+ provider registry.terraform.io/scaleway/scaleway v2.41.3
```

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* scaleway_iam_application

### Output

<!--- If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`. --->
```
Stack trace from the terraform-provider-scaleway_v2.41.3 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x10f4e55]

goroutine 220 [running]:
github.com/scaleway/terraform-provider-scaleway/v2/internal/services/secret.datasourceSchemaFromResourceVersionSchema({0x197a070, 0xc000b0c0e0}, 0xc000f32880, {0x130c960, 0xc0007a3020})
	github.com/scaleway/terraform-provider-scaleway/v2/internal/services/secret/version_data_source.go:88 +0x7b5
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc000713b20, {0x1979fc8, 0xc0005471a0}, 0xc000f32880, {0x130c960, 0xc0007a3020})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:823 +0x119
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).ReadDataApply(0xc000713b20, {0x1979fc8, 0xc0005471a0}, 0xc000f32780, {0x130c960, 0xc0007a3020})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:1043 +0x13a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadDataSource(0xc00012d158, {0x1979fc8?, 0xc0004bd140?}, 0xc000f01d40)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/grpc_provider.go:1434 +0x6b1
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ReadDataSource(0xc00029b960, {0x1979fc8?, 0xc000f01da0?}, 0xc000f01d40)
	github.com/hashicorp/terraform-plugin-mux@v0.16.0/tf5muxserver/mux_server_ReadDataSource.go:36 +0x193
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadDataSource(0xc0006183c0, {0x1979fc8?, 0xc000f015c0?}, 0xc000896e10)
	github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/tf5server/server.go:688 +0x27d
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadDataSource_Handler({0x13909e0, 0xc0006183c0}, {0x1979fc8, 0xc000f015c0}, 0xc000f32580, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:572 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00021ee00, {0x1979fc8, 0xc000f01530}, {0x197fa60, 0xc000235b00}, 0xc00091a5a0, 0xc0006a2e10, 0x2280000, 0x0)
	google.golang.org/grpc@v1.63.2/server.go:1369 +0xdf8
google.golang.org/grpc.(*Server).handleStream(0xc00021ee00, {0x197fa60, 0xc000235b00}, 0xc00091a5a0)
	google.golang.org/grpc@v1.63.2/server.go:1780 +0xe8b
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.63.2/server.go:1019 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 55
	google.golang.org/grpc@v1.63.2/server.go:1030 +0x125

Error: The terraform-provider-scaleway_v2.41.3 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

```

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform import scaleway_iam_application.main xxx`

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

* #2450

--- a/internal/services/secret/version_data_source.go
+++ b/internal/services/secret/version_data_source.go
@@ -73,19 +73,17 @@ func datasourceSchemaFromResourceVersionSchema(ctx context.Context, d *schema.Re
 			return diag.FromErr(err)
 		}
 
-		secretByName := (*secret.Secret)(nil)
-		for _, s := range secrets.Secrets {
-			if s.Name == secretName {
-				if secretByName != nil {
-					return diag.Errorf("found multiple secret with the same name (%s)", secretName)
-				}
-				secretByName = s
-			}
+		foundSecret, err := datasource.FindExact(secrets.Secrets,
+			func(s *secret.Secret) bool { return s.Name == secretName },
+			secretName,
+		)
+		if err != nil {
+			return diag.FromErr(err)
 		}
 
 		res, err := api.AccessSecretVersion(&secret.AccessSecretVersionRequest{
 			Region:   region,
-			SecretID: secretByName.ID,
+			SecretID: foundSecret.ID,
 			Revision: d.Get("revision").(string),
 		}, scw.WithContext(ctx))
 		if err != nil {

